src/connectors/hierarchical-menu/connectHierarchicalMenu.ts

```diff
@@ -174,7 +181,8 @@ const connectHierarchicalMenu: HierarchicalMenuConnector = function connectHiera
       limit = 10,
       showMore = false,
       showMoreLimit = 20,
-      sortBy = ['name:asc'],
+      sortBy = DEFAULT_SORT,
+      facetOrdering = sortBy === DEFAULT_SORT,
```

Comment:
Here the `sortBy === DEFAULT_SORT` portion is evaluated when `facetOrdering` has not yet been declared into the scope, but it assumes its sibling `sortBy` has been.
For example, someone refactoring would think it's good a idea to have theses attributes alpha sorted. It's an object literal, `{a, b}` or `{b, a}`, it shouldn't matter.

But then this doesn't work:

```js
{ 
   facetOrdering = sortBy === DEFAULT_SORT,
   ...
   sortBy = DEFAULT_SORT
}
```

So I think it's easy to mess up. The kind of things that would go into a _ES6 the bad part_ 😄
WDYT?

Reply:
True, but not sure what the alternative is? A separate declaration?

Reply:
yes that's why I had in mind. I know it's not as pretty
```js
{
   sortBy = DEFAULT_SORT,
   facetOrdering
}

if (facetOrdering === undefined) {
    facetOrdering = (sortBy === DEFAULT_SORT)
}

```

Reply:
not really a fan of that, since it requires two different declarations and to make facetOrdering a `let` or rename the variable for a condition that works totally fine. If facetOrdering is moved on top of sortBy, eslint will warn about "use before define" :)

Reply:
Yes and typescript won't compile it so I guess it's fine after all.